File: web/src/api/auth.api.ts
import { http } from "./http"
import type { ApiResponse, AuthResponse, LoginRequest, RegisterRequest } from "@/types/auth";

export const authApi = {
    async register(payload: RegisterRequest): Promise<ApiResponse> {
        const { data } = await http.post<ApiResponse>("/api/auth/register", payload);
        return data;
    },

    async login(payload: LoginRequest): Promise<AuthResponse> {
        const { data } = await http.post<AuthResponse>("/api/auth/login", payload);
        return data;
    },

    async logout(): Promise<ApiResponse> {
        const { data } = await http.post<ApiResponse>("/api/auth/logout");
        return data;
    },
}

File: mobile/app/src/main/java/com/revilleza/userauth/network/api/AuthApi.kt
package com.revilleza.userauth.network.api

import com.revilleza.userauth.network.dto.AuthResponse
import com.revilleza.userauth.network.dto.LoginRequest
import com.revilleza.userauth.network.dto.RegisterRequest
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthApi {

    @POST("/api/auth/register")
    suspend fun register(@Body req: RegisterRequest): AuthResponse

    @POST("/api/auth/login")
    suspend fun login(@Body req: LoginRequest): AuthResponse
}